src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -252,25 +252,27 @@ private List<string> CreateListOfParameterValues()
             }
 
             // Add parameters on any output tags
-            foreach (ProjectTaskInstanceChild taskOutputSpecification in _taskNode.Outputs)
+            // Perf: Moved to indexed based for loop to avoid boxing of the enumerator for IList.
+            for (int i = 0; i < _taskNode.Outputs.Count; i++)
             {
-                ProjectTaskOutputItemInstance outputItemInstance = taskOutputSpecification as ProjectTaskOutputItemInstance;
+                var projectTaskInstanceChild = _taskNode.Outputs[i];
```

Comment:
extreme nit: if you instead named this `taskOutputSpecification` you could have had a smaller diff.